src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -398,6 +412,19 @@ def generate_tool_meta(
                 tool_dict[source] = generate_tool_meta_dict_by_file(source, tool_type)
             except Exception as e:
                 exception_dict[source] = ExceptionPresenter.create(e).to_dict()
+    return tool_dict, exception_dict
+
+
+def _generate_tool_meta_and_update_dict(
+    working_dir: Path,
+    tools: Mapping[str, Mapping[str, str]],
+    tool_dict: dict,
+    exception_dict: dict,
+    prevent_terminate_signal_propagation: bool = False,
+):
+    _tool_dict, _exception_dict = generate_tool_meta(working_dir, tools)
```

Comment:
Pass `prevent_terminate_signal_propagation` to generate_tool_meta

Reply:
good catch, updated